service/src/intTest/java/uk/nhs/adaptors/gp2gp/gpc/GpcFindDocumentsComponentTest.java

```diff
@@ -0,0 +1,114 @@
+package uk.nhs.adaptors.gp2gp.gpc;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.UUID;
+
+import uk.nhs.adaptors.gp2gp.common.task.BaseTaskTest;
+import uk.nhs.adaptors.gp2gp.ehr.EhrExtractStatusRepository;
+import uk.nhs.adaptors.gp2gp.ehr.EhrExtractStatusTestUtils;
+import uk.nhs.adaptors.gp2gp.ehr.EhrStatusConstants;
+import uk.nhs.adaptors.gp2gp.ehr.model.EhrExtractStatus;
+import uk.nhs.adaptors.gp2gp.testcontainers.ActiveMQExtension;
+import uk.nhs.adaptors.gp2gp.testcontainers.MongoDBExtension;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+@ExtendWith({SpringExtension.class, MongoDBExtension.class, ActiveMQExtension.class})
+@SpringBootTest
+@DirtiesContext(classMode = DirtiesContext.ClassMode.AFTER_EACH_TEST_METHOD)
+public class GpcFindDocumentsComponentTest extends BaseTaskTest {
+    private static final String NHS_NUMBER_WITH_DOCUMENT = "9690937286";
+    private static final String NHS_NUMBER_WITHOUT_DOCUMENT = "9690937294";
+    private static final String VALID_DOCUMENT_URL = "https://orange.testlab.nhs.uk/B82617/STU3/1/gpconnect/documents/fhir/Binary/07a6483f-732b-461e-86b6-edb665c45510";
+
+    @Autowired
+    private GpcFindDocumentsTaskExecutor gpcFindDocumentsTaskExecutor;
+    @Autowired
+    private EhrExtractStatusRepository ehrExtractStatusRepository;
+
+    private EhrExtractStatus setupDatabase() {
+        var ehrExtractStatus = EhrExtractStatusTestUtils.prepareEhrExtractStatus();
+        ehrExtractStatus.setGpcAccessDocument(EhrExtractStatus.GpcAccessDocument.builder()
+            .documents(List.of(
+                EhrExtractStatus.GpcAccessDocument.GpcDocument.builder()
+                    .documentId(EhrStatusConstants.DOCUMENT_ID)
+                    .accessDocumentUrl(EhrStatusConstants.GPC_ACCESS_DOCUMENT_URL)
+                    .build()
+            ))
+            .build());
+        return ehrExtractStatusRepository.save(ehrExtractStatus);
+    }
+
```

Comment:
Add a test to confirm data is overwritten

Reply:
Added test after the method is called.